util/setuidgid.c

```diff
@@ -0,0 +1,360 @@
+/* We use non-POSIX functionality in this file. */
+#undef _POSIX_C_SOURCE
+#undef _XOPEN_SOURCE
+
+/*
+ * There is no setgroups() in the POSIX standard, so if we want to drop
+ * supplementary groups, we need to use platform-specific code.  This must
+ * happen before the regular includes, as in some cases we need to define other
+ * symbols before including the relevant header.
+ */
+#if defined(__linux__)
+/* setgroups() includes for Linux. */
+#define _BSD_SOURCE 1
+#include <grp.h>
+#define HAS_SETGROUPS 1
```

Comment:
Rather than having lots of `#define HAS_SETGROUPS 1` lines, wouldn't it be easier to have a single `#define NO_SETGROUPS` at the end?